File: counter_publisher_py/src/counter_publisher_py/counter_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class CounterPublisher(Node):
    def __init__(self):
        super().__init__('counter_publisher_py')
        self.publisher_ = self.create_publisher(String, '/counter', 10)
        self.timer = self.create_timer(1.0, self.publish_count)
        self.count_ = 0
        self.get_logger().info('Publisher Python iniciado, publicando en /counter')

    def publish_count(self):
        msg = String()
        msg.data = f'py: {self.count_}'
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publicando: {msg.data}')
        self.count_ += 1

def main(args=None):
    rclpy.init(args=args)
    node = CounterPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == '__main__':
    main()

File: counter_publisher_cpp/src/counter_publisher.cpp
#include <chrono>
#include <memory>
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <sstream>

using namespace std::chrono_literals;

class CounterPublisher : public rclcpp::Node {
public:
  CounterPublisher() : Node("counter_publisher_cpp"), count_(0) {
    publisher_ = create_publisher<std_msgs::msg::String>("/counter", 10);
    timer_ = create_wall_timer(
        1s, std::bind(&CounterPublisher::publish_count, this));
    RCLCPP_INFO(this->get_logger(), "Publisher iniciado, publicando en /counter");
  }

private:
  void publish_count() {
    auto message = std_msgs::msg::String();
    std::stringstream ss;
    ss << "c++: " << count_++;
    message.data = ss.str();

    RCLCPP_INFO(this->get_logger(), "Publicando: %s", message.data.c_str());
    publisher_->publish(message);
  }

  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  int count_;
};

int main(int argc, char *argv[]) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<CounterPublisher>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
